mbuild/formats/hoomd_snapshot.py

```diff
@@ -5,15 +5,79 @@
 import numpy as np
 import parmed as pmd
 
-import mbuild as mb
+from mbuild.box import Box
+from mbuild.compound import Compound, Particle
 from mbuild.utils.geometry import coord_shift
 from mbuild.utils.io import import_
 from mbuild.utils.sorting import natural_sort
 
 hoomd = import_("hoomd")
 hoomd.data = import_("hoomd.data")
 
-__all__ = ["to_hoomdsnapshot"]
+__all__ = ["to_hoomdsnapshot", "from_snapshot"]
+
+
+def from_snapshot(snapshot, scale=1.0):
+    """Convert a Snapshot to a Compound.
+
+    Snapshot can be a hoomd.data.Snapshot or a gsd.hoomd.Snapshot.
+
+    Parameters
+    ----------
+    snapshot : hoomd._hoomd.SnapshotSystemData_float or gsd.hoomd.Snapshot
+        Snapshot from which to build the mbuild Compound.
+    scale : float, optional, default 1.0
+        Value by which to scale the length values
+
+    Returns
+    -------
+    comp : Compound
+
+    Note
+    ----
+    GSD and HOOMD snapshots center their boxes on the origin (0,0,0), so the
+    compound is shifted by half the box lengths
+    """
+    comp = Compound()
+    bond_array = snapshot.bonds.group
+    n_atoms = snapshot.particles.N
+
+    # There will be a better way to do this once box overhaul merged
```

Comment:
We can remove this line here since the box overhaul is merged (unless there are additional changes planned in the near future). Potentially add a #TODO if we want future changes to our box class to be aligned with the snaphsot.box.